src/perturbations.jl

```diff
@@ -64,23 +107,45 @@ function hierarchy!(du, u, hierarchy::Hierarchy{T, BasicNewtonian}, x) where T
 
     # relativistic neutrinos (massless)
     𝒩′[0] = -k / ℋₓ * 𝒩[1] - Φ′
-    𝒩′[1] = k/(3ℋₓ)*𝒩[0] - 2*k/(3ℋₓ)*𝒩[2] + k/(3ℋₓ)*Ψ
+    𝒩′[1] = k/(3ℋₓ)*𝒩[0] - 2*k/(3ℋₓ) *𝒩[2] + k/(3ℋₓ) *Ψ
     for ℓ in 2:(ℓ_ν-1) #ℓ_ν same as ℓᵧ for massless nu for now
-        𝒩′[ℓ] =  k / ((2ℓ+1) * ℋₓ) *( ℓ*𝒩[ℓ-1] - (ℓ+1)*𝒩[ℓ+1])
+        𝒩′[ℓ] =  k / ((2ℓ+1) * ℋₓ) * ( ℓ*𝒩[ℓ-1] - (ℓ+1)*𝒩[ℓ+1] )
     end
-    #truncation
-    𝒩′[ℓ_ν] =  k / ℋₓ  * 𝒩[ℓ_ν-1] - (ℓ_ν+1)/(ℋₓ *ηₓ) *𝒩[ℓ_ν]#Callin 06
-
-    # WIP: massive nu
-    # # neutrinos (massive, MB 57) - change convention
-    # #units not yet right
-    # Ψ_ν′[0] = -q*k / ϵ * Ψ_ν[1] - Φ′ *dnlnf0dlnq #FIXME dnln, def Psi,IC, Einstein int
-    # Ψ_ν′[1] = q*k/(3 ϵ)*(𝒩[0] - 2*Ψ_ν[2]) - This k*ϵ / (3*q)*Ψ *dnlnf0dlnq
-    # I think can't mutate the u variable...
-    # Ψ_ν[ℓ_νm] = (2*ℓ_νm+1)*ϵ/(q*k*ηₓ)*Ψ_ν[ℓ_νm] - Ψ_ν[ℓ_νm-1] #truncation of MB (51)
-    # for ℓ in 2:(ℓ_νm-1) #ℓ_νm should be smaller than massless case
-    #     Ψ_ν′[ℓ] =  q*k / ((2ℓ+1) * ϵ) *( ℓ*Ψ_ν[ℓ-1] - (ℓ+1)*Ψ_ν[ℓ+1])
-    # end
+    #truncation (same between MB and Callin06)
+    𝒩′[ℓ_ν] =  k / ℋₓ  * 𝒩[ℓ_ν-1] - (ℓ_ν+1)/(ℋₓ *ηₓ) *𝒩[ℓ_ν]
+
+    #WIP: nonrelativistic nu
+    # neutrinos (massive, MB 57)
+    for (i_q, q) in zip(Iterators.countfrom(0), q_pts)
```

Comment:
Wow, the neutrinos add a lot of variables to the ODE (+160 or more), and they are super sparse since they only interact with the metric. We'll need to eventually incorporate sparsity considerations since (to my recollection) implicit ODE methods scale like n^3  (@marius311).

Reply:
Just a quick note - I did not try changing up the number of multipoles and only did for nq a little - so not sure how many are totally necessary. I think we should drop \ell_m\nu to be smaller than \ell_\nu since the higher-\ell moments become unimportant as neutrinos become nonrelativistic.

Reply:
What we want is something like https://diffeq.sciml.ai/stable/tutorials/advanced_ode_example/#Automatic-Sparsity-Detection but it seems like this is now in ModelingToolkit? I need to spend a weekend really looking into this.

Reply:
Ok, tried some sparsity detection. We're still not at the level where using the sparse matrix type makes sense for us!